runtime/js/10_dispatch_buffer.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+
+  function assert(cond) {
+    if (!cond) {
+      throw Error("assert");
+    }
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////// General async handling //////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  // General Async response handling
+  let nextRequestId = 1;
+  const promiseTable = {};
+
+  function opAsync(opName, opRequestBuilder, opResultParser) {
+    // Make sure requests of this type are handled by the asyncHandler
+    // The asyncHandler's role is to call the "promiseTable[requestId]" function
+    core.setAsyncHandlerByName(opName, (bufUi8, _) => {
+      const [requestId, result, error] = opResultParser(bufUi8, true);
+      if (error !== null) {
+        promiseTable[requestId][1](error);
+      } else {
+        promiseTable[requestId][0](result);
+      }
+      delete promiseTable[requestId];
+    });
+
+    const requestId = nextRequestId++;
+
+    // Create and store promise
+    const promise = new Promise((resolve, reject) => {
+      promiseTable[requestId] = [resolve, reject];
+    });
+
+    // Synchronously dispatch async request
+    core.dispatchByName(opName, ...opRequestBuilder(requestId));
+
+    // Wait for async response
+    return promise;
+  }
+
+  function opSync(opName, opRequestBuilder, opResultParser) {
+    const rawResult = core.dispatchByName(opName, ...opRequestBuilder());
+
+    const [_, result, error] = opResultParser(rawResult, false);
+    if (error !== null) throw error;
+    return result;
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  /////////////////////////////////// Error handling /////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  function handleError(className, message) {
+    const [ErrorClass, args] = core.getErrorClassAndArgs(className);
+    if (!ErrorClass) {
+      return new Error(
+        `Unregistered error class: "${className}"\n  ${message}\n  Classes of errors returned from ops should be registered via Deno.core.registerErrorClass().`,
```

Comment:
This line is longer than 80 characters, it's strange that formatter doesn't complain about it

Reply:
In the latest commit the string should be split.